[czhiruo] iP

czhiruo opened the pull request:
### PAU CHATBOT

Looking for a chatbot that is

- [x] your personal assistant, 
- [x] your virtual sidekick, 
- [x] and your ultimate taskmaster?

Meet PAU, your **adorable** digital companion in the world of productivity🤩! Pau is not just any chatbot; Pau 

- keeps track of your tasks _effortlessly_
- helps you stay on top of your deadlines
- and is your trusted calendar

> "Procrastination is the thief of time" -Edward Young

Here's how you can get a PAU:
1. Download PAU [here](https://github.com/czhiruo/ip)
2. Give PAU some tasks
3. PAU is ready to help you😍

Here's a look into Pau's `main` method:
```java
    public static void main(String[] args) {
        new Pau("./data/paulist.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

czhiruo committed:
completed level 0

czhiruo committed:
completed level 0

czhiruo committed:
completed level 0

czhiruo committed:
completed level 1

czhiruo committed:
no message

czhiruo committed:
completed level 2

czhiruo committed:
completed level 3

czhiruo committed:
completed level 4

czhiruo committed:
completed level 5

czhiruo committed:
completed level 6

czhiruo committed:
completed Automated Text UI Testing

czhiruo committed:
added javadocs

czhiruo committed:
Rename the file

czhiruo committed:
Format string output

czhiruo committed:
Complete level 7

czhiruo committed:
Complete level 7

czhiruo committed:
Complete level 7

czhiruo committed:
Create TaskList class

czhiruo committed:
Complete Level 8

czhiruo committed:
Add More OOP

czhiruo committed:
Add to package

czhiruo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

czhiruo committed:
Add JUnit Tests

czhiruo committed:
Add Javadocs

czhiruo committed:
Format code to coding standard

czhiruo committed:
Complete Level 9

czhiruo committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/test/java/pau/task/TaskListTest.java
#	src/test/java/pau/util/UiTest.java

czhiruo committed:
Complete Level 9

czhiruo committed:
Merge branch 'branch-Level-9'

czhiruo committed:
Fix Errors

DeltonCheng commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+/**
+ * Stores the tasks in an ArrayList;
+ */
+public class TaskList {
```
Maybe you could let `Ui.java` handle the lines that are being printed?

DeltonCheng commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+     *
+     * @param input The task the user wants to unmark.
+     */
+    public void unMarkTask(String input) {
```
Might be nitpicking, but should the naming be unmarkTask instead? Since unmark is a single word

DeltonCheng commented on src/main/java/pau/util/Parser.java:
```diff
@@ -0,0 +1,37 @@
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
```
Parser is very clean, good job!

DeltonCheng commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+     */
+    public void markTask(String input) {
+        int starEyesEmoji = 0x1F929;
+        System.out.println("good job, you've completed a task! You're so productive!" + new String(Character.toChars(starEyesEmoji)));
```
should wrap this line, might be too long

DeltonCheng reviewed (commented):
Really nice and clean code! Just note some minor tweaks for clarity~ Good job! :)

oeggy03 commented on src/main/java/pau/exception/DeadlineNoEndException.java:
```diff
@@ -0,0 +1,18 @@
+    public DeadlineNoEndException(String errorMessage) {
+        int cryingEmojiUnicode = 0x1F62D;
+
+        System.out.println("when is this due" + new String(Character.toChars(cryingEmojiUnicode)));
```
Maybe you can have a separate printer class to handle all printing for the bot

oeggy03 commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+        Task checkedTask = this.taskList.get(taskNo - 1);
+
+        checkedTask.markAsUndone();
+        System.out.println("why are you not going to " + checkedTask.description + "? remember to do it later!");
```
Might lead to grammatically weird sentences if you print it with the description lol (small nitpick)

oeggy03 commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+            this.taskList.remove(checkedTask);
+            if (checkedTask.getStatusIcon().equals("X")) {
+                System.out.println("good job! you're officially done with this:");
+                System.out.println(checkedTask.toString());
```
you can lump the 2 System.out.println into 1, might make the code cleaner

oeggy03 commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+     *
+     * @param input The ToDo the user wants to add.
+     */
+    public void addToDo(String input) {
```
Maybe instead of having an add method for each task type, you can have a general addtask method here, and abstract the specifics in each class (Todo, Deadline etc.)

oeggy03 commented on src/main/java/pau/util/Parser.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    public static void parseCommand(String input, TaskList list) {
+        if (input.equals("list")) {
+            list.checkList();
```
Can create separate classes for each command parser, better for oop

oeggy03 commented on src/test/java/pau/util/UiTest.java:
```diff
@@ -0,0 +1,57 @@
+        Ui ui = new Ui();
+
+        String expected = " HI! I'm \n" +
+                "██████╗  █████╗ ██╗   ██╗\n" +
```
omg this is nice

oeggy03 reviewed (commented):
LGTM I like the OOP and package structure, very nice

czhiruo committed:
Set Up GUI

zekone commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+    /**
+     * Prints out a list of the tasks the user has.
+     */
+    public void checkList() {
```
can consider renaming to `printChecklist` to improve clarity since the expected behavior is to print?

zekone commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+     */
+    public void checkList() {
+        if (this.listSize() == 0) {
+            int starEyesEmoji = 0x1F929;
```
really like that you are using emoji to make the chatbot more interesting.

zekone commented on src/main/java/pau/util/Ui.java:
```diff
@@ -0,0 +1,34 @@
+    /**
+     * Name of the chatbot.
+     */
+    private final String botName = "\n" +
+            "██████╗  █████╗ ██╗   ██╗\n" +
+            "██╔══██╗██╔══██╗██║   ██║\n" +
+            "██████╔╝███████║██║   ██║\n" +
+            "██╔═══╝ ██╔══██║██║   ██║\n" +
+            "██║     ██║  ██║╚██████╔╝\n" +
+            "╚═╝     ╚═╝  ╚═╝ ╚═════╝ \n" +
+            "                         \n";
```
really creative, I like it a lot!

zekone commented on src/main/java/pau/task/TaskList.java:
```diff
@@ -0,0 +1,247 @@
+     */
+    public void deleteTask(String input) {
+        try {
+            String parts[] = input.split(" ");
+            int taskNo = Integer.parseInt(parts[1]);
+            if (taskNo > this.listSize()) {
+                throw new NoSuchTaskException("you can't delete this task");
+            }
+            Task checkedTask = this.taskList.get(taskNo - 1);
+            this.taskList.remove(checkedTask);
+            if (checkedTask.getStatusIcon().equals("X")) {
```
nitpicking here, but maybe add line breaks to make it more readable?

zekone commented on src/main/java/pau/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+     * @param status String of an integer that represents state of the task.
+     */
+    public void setStatus(String status) {
+        if ((status == "1")) {
```
why is there double brackets here?

zekone commented on src/main/java/pau/task/Task.java:
```diff
@@ -0,0 +1,75 @@
+        this.isDone = false;
+    }
+
+    /**
+     * Writes the contents of the task to the hard disk.
+     *
+     * @return The contents of the task that is saved to the hard disk.
+     */
+    public abstract String writeToFile();
```
I think the description of this method may be misleading? From what I have read in the `ToDo`, `Deadline` and `Event` implementation of the method, it is simply returning the storage format in a `String` and not actually writing to the file? Can consider renaming the method to `getStorageFormat` and update the description to make it clearer?

zekone reviewed (commented):
LGTM

czhiruo committed:
Implement Checkstyle

czhiruo committed:
Complete GUI

czhiruo committed:
Exit Chatbot For Bye Command

czhiruo committed:
Add Assertion

czhiruo committed:
Improve Code Quality

There was no help command.

Users would not be able to ask for help to create tasks.

A help command is added that lists out how to execute the different commands.

czhiruo committed:
Merge pull request #2 from czhiruo/branch-A-Assertions

Add Assertion

czhiruo committed:
Merge pull request #3 from czhiruo/branch-A-CodeQuality

Improve Code Quality

czhiruo committed:
Update README.md

czhiruo committed:
Improve UI and Add Clear Command

Added background colours for user and pau dialog boxes.

Added title for the stage.

Added a clear command to clear all the tasks in the list.

czhiruo committed:
Update README.md

czhiruo committed:
Update README.md

czhiruo committed:
Update README.md

czhiruo committed:
Update README.md

czhiruo committed:
Update README.md

czhiruo committed:
Update README.md

czhiruo committed:
Update UI

czhiruo committed:
Add Ui.png via upload

czhiruo committed:
Fix Tests and Catch Exceptions

czhiruo committed:
Fix Tests and Catch Exceptions

czhiruo committed:
Fix Code Breaking
